compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -2226,9 +2227,10 @@ impl<'a, 'tcx> InferCtxtPrivExt<'a, 'tcx> for InferCtxt<'a, 'tcx> {
         post.dedup();
 
         if self.is_tainted_by_errors()
-            && crate_names.len() == 1
-            && ["`core`", "`alloc`", "`std`"].contains(&crate_names[0].as_str())
-            && spans.len() == 0
+            && (crate_names.len() == 1
```

Comment:
Could this check just be simplified to "if the self type is a numeric infer"?

Reply:
So we need to check for int/float variables not just in self, but in the trait's substs as well, so we don't report the things that implement `ConcreteType: Trait<{integer}>`. 

Also removing the core/alloc/std crate name check leads to some regressions in errors reported with simd stuff and when we're unable to infer const variables due to other errors :/